components/brave_rewards/core/wallet_provider/gemini/connect_gemini_wallet.cc

```diff
@@ -194,6 +196,12 @@ void ConnectGeminiWallet::OnPostAccount(ConnectExternalWalletCallback callback,
         base::unexpected(mojom::ConnectExternalWalletError::kUnexpected));
   }
 
+  // Update the user's "declared country" based on the information provided by
+  // Gemini.
+  if (!country_id.empty()) {
+    engine_->SetState(state::kDeclaredGeo, country_id);
+  }
```

Comment:
This should go to `ConnectExternalWallet::OnConnect`, so that we only have to do it once.

Reply:
👍